### PR TITLE
Feat/suggestion testing

### PR DESCRIPTION
**Pull request by rensdoodeman**
suggestion e2e testing and added error-handling + spinner

### CODE REVIEW

**Comment by maarcelh** [comment]
Kan je dev erin mergen?

**Review by KevinBreurken — APPROVED** [review]

**Comment by fkoulen** [comment]
@rensdoodeman why was this merged even tho it failed deploying? lol

File: cypress/e2e/suggestion.cy.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const suggestionId = Cypress.env("CYPRESS_SUGGESTION_ID")
+
+describe("suggestion detail page", () => {
```
Comment (fkoulen) [inline cypress/e2e/suggestion.cy.ts:3]: @rensdoodeman the main functionality of this page is entering what instrument you plan on playing, yet there seems to be missing any tests for this. Are you considering writing those tests?

File: cypress/e2e/suggestions.cy.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+describe("suggestions page", () => {
```
Comment (fkoulen) [inline cypress/e2e/suggestions.cy.ts:1]: There are no tests here for edge cases such as a failing requests, the loading spinner or when there are no requests. I think that you should not only test that it works, but also test what happens when it does not work.

i.e. you could stimulate no responses with this:
`cy.intercept("/rest/v1/suggestion*", [])`

File: src/pages/suggestions/index.tsx [hunk 3]
```diff
@@ -6,49 +6,61 @@ import { useSupabaseClient } from "@supabase/auth-helpers-react"
...
+        </div>}
+
+      {showLoadingError && <div className={"mt-6"}>
+        <ErrorPopup text={"Failed to load suggestions."} closePopup={() => setShowLoadingError(false)} />
+      </div>}
+
+      {noSuggestionsMade && <div className={"flex max-w-m gap-4 text-zinc-400 items-center justify-center"}>
+        <div><MagnifyingGlassCircleIcon className={"w-[50px] h-[50px]"} /></div>
+        <p>Looks like there are no suggestions made yet! Feel free to start adding them.</p>
+      </div>}
```
Comment (fkoulen) [inline src/pages/suggestions/index.tsx:63]: This is always shown when it is loading the suggestions. I think you want to hide it then as well.
